src/operators.jl

```diff
@@ -309,3 +274,104 @@ function tensor_product{G<:SimpleGraph}(g::G, h::G)
     end
     return z
 end
+
+
+## subgraphs ###
+
+"""
+    induced_subgraph(g, vlist)
+
+Returns the subgraph of `g` induced by the vertices in  `vlist`.
+
+The returned graph has `length(vlist)` vertices, with the new vertex `i`
+corresponding to the vertex of the original graph in the `i`-th position
+of `vlist`.
+
+Returns  also a vector `vmap` mapping the new vertices to the
+old ones: the  vertex `i` in the subgraph corresponds to
+the vertex `vmap[i]` in `g`.
+
+    induced_subgraph(g, elist)
+
+Returns the subgraph of `g` induced by the edges in `elist`, along with
+the associated vector `vmap` mapping new vertices to the old ones.
+
+
+### Usage Examples:
+```julia
+g = CompleteGraph(10)
+sg, vmap = subgraph(g, 5:8)
+@assert g[5:8] == sg
+@assert nv(sg) == 4
+@assert ne(sg) == 6
+@assert vm[4] == 8
+
+sg, vmap = subgraph(g, [2,8,3,4])
+@asssert sg == g[[2,8,3,4]]
+
+elist = [Edge(1,2), Edge(3,4), Edge(4,8)]
+sg, vmap = subgraph(g, elist)
+@asssert sg == g[elist]
+```
+"""
+function induced_subgraph{T<:SimpleGraph}(g::T, vlist)
```

Comment:
The common supertype is now `AbstractArray{Int64,1}` - this was originally omitted since there was no common supertype. I realize we're making a new change, but let's add the annotation.